### ISSUE TITLE
[Feature] Recipe for plupload v2

### ISSUE THREAD

**Issue by burzum** [abramenal/cypress-file-upload#160]
### Current behavior:

We still use plupload v2 in some places. Now I've tried to add tests to these pages without much success. If I'm doing the same procedure manually the very same file just uploads fine. I've pasted our test, our custom filter code and a screenshot below. I really can't figure out what the difference is in the test.

```js
import 'cypress-file-upload';

describe('Projects', function () {

	before(function () {
		cy.loadAllFixtures();
	});

	beforeEach(function () {
		cy.loadFixtures(['app.Projects']);
	});

	it('Create a new project', function () {
		cy.login('manufacturer@test.com', 'password');
		cy.visit('/en/profiles/edit/projects');
		cy.get('.list-unstyled > :nth-child(2) > .btn').click({force: true});

		// Assert redirect
		cy.url().should('match', /\/en\/projects\/edit\/8/);

		let filename = '1600x1200.jpg';
		cy.fixture('files/' + filename, 'base64').then(file => {
			cy.get('input[type="file"]').upload({
				fileContent: file.toString(),
				fileName: filename,
				mimeType: 'image/jpg',
				encoding: 'base64'
			});
		});
	});
});
```

Our custom filter we use to validate the image size:

```js
plupload.addFileFilter('min_img_resolution', function(maxRes, file, cb) {
	if (file.type === 'image/svg+xml') {
		let img = new window.moxie.image.Image();

		function finalize(result) {
			// cleanup
			img.destroy();
			img = null;

			cb(true);
		}

		img.onload = function() {
			finalize(true);
		};

		img.onerror = function() {
			finalize(false);
		};

		img.load(file.getSource());
	} else {
		let self = this;
		let img = new window.moxie.image.Image();

		function finalize(result) {
			// cleanup
			img.destroy();
			img = null;

			// if rule has been violated in one way or another, trigger an error
			if (!result) {
				self.trigger('Error', {
					code: plupload.IMAGE_DIMENSIONS_ERROR,
					message: 'Image should be at least ' + maxRes[0] + ' pixels wide.',
					file: file
				});
			}
			cb(result);
		}

		img.onload = function() {
			if (img.type === 'svg') {
				finalize(true);
			} else {
				finalize(img.width >= maxRes[0] && img.height >= maxRes[1]);
			}
		};

		img.onerror = function() {
			finalize(false);
		};

		img.load(file.getSource());
	}
});
```

![image](https://user-images.githubusercontent.com/162789/74891430-516cda00-5387-11ea-9c45-44818ad61a4a.png)

### Desired behavior:

The image passing the upload.

### Versions

Cypress 4.0.2, Chrome 80.0.3987.116

**Comment by abramenal** [comment]
Hi @burzum 
Thanks for submitting the issue!

Have you tried that exact code without plupload validation? Did it work?
I am thinking if cypress could modify image's metadata..

Please let me know if that work. I think I'll have some time tomorrow to debug this.

**Comment by burzum** [comment]
@abramenal thanks for taking care of it. The issue is indeed something within the validation. When I remove the whole rule it uploads the file. The weird thing is, it works on our live site and in my dev environment as well. `file.type` is image/jpg so the type is OK. There must be something else that doesn't work well with this Moxie library but it is a pain to debug. 😞

**Comment by abramenal** [comment]
@burzum this seems to be expected behavior:

> Width of the image. Actual value is set only after image is preloaded.

https://github.com/moxiecode/moxie/wiki/moxie.image.Image#width

**Comment by burzum** [comment]
@abramenal sure but why is it working when I'm not using Cypress but working when doing it manually? I can't really figure out how and were to modify the fixture or data.

I've already put some debug code into my filter.

```js
		img.onload = function() {
			if (img.type === 'svg') {
				finalize(true);
			} else {
				console.log('onload');
				console.log(file);
				finalize(img.width >= maxRes[0] && img.height >= maxRes[1]);
			}
		};
```

But it seems to never go into this part. I've also tried to add `img.preload = function() {...}` to debug it but this seems to never get called either. :(

**Comment by burzum** [comment]
I'm running the very same code side by side in the browser opened by Cypress and it clearly behaves different when loading the file via Cypress (left) and doing it manually (right):

![image](https://user-images.githubusercontent.com/162789/75625333-71896e00-5bbd-11ea-9567-f3bf33688696.png)

**Comment by abramenal** [comment]
@burzum can you please create a sample repl with such component so I can investigate?
I am almost done with v4 source code and want to test your stuff there.

**Comment by burzum** [comment]
@abramenal I'll do so tomorrow!

**Comment by burzum** [comment]
@abramenal I've extracted the plupload to a new repository that includes our upload filter as well that validates the image. For whatever reason, when extracted to this lib it even fails for me if I do it manually. In our application it works as expected... We decided to delay Cypress tests for pages with the upload and remove plupload because it seems to be unmainted and outdated and switch to https://www.dropzonejs.com/. I hope its easier to get this one to work with Cypress.

However, I've pushed what I have to https://github.com/burzum/cypress-plupload just in case you still want to take a look at it.
